Assets/Source/Components/Player/PlayerBehavior.cs

```diff
@@ -150,15 +236,20 @@ private void FireBlaster()
 
         private void UpdateActorStatus()
         {
-            // todo: this is just placeholder stuff
-            if (actorBehavior.Health <= 0)
-            {
-                InstantiateInLevel(explosionPrefab, transform.position);
-                menuSelector.ShowMenu<GameOverMenuComponent>();
-                Destroy(gameObject);
-            }
+            //this has to be done here rather than actorBehavior
+            //cuz actorbehavior is also used by other instances so we ended up updating the hud from the enemy
+            ammoText.SetText($"x{actorBehavior.BlasterAmmo}");
```

Comment:
this also should be moved to a UI component class.  We should not couple our player to the UI

Reply:
decoupled